awx/main/access.py

```diff
@@ -1397,6 +1397,8 @@ def changes_are_non_sensitive(self, obj, data):
         ]
 
         for k, v in data.items():
+            if k not in [x.name for x in obj._meta.concrete_fields]:
```

Comment:
Depending on the length of `data.items()` and `obj._meta_concrete_fields` this can be a very time expensive task since the list will have to be evaluated over and over again. If you think this is going to be an issue, then you could create a cache. Something like:

```python
obj_concrete_fields = [x.name for x in obj._meta.concrete_fields]
for k, v in data.items():
    if k not in obj_concrete_fields:
        continue
    # ...
```

Reply:
I don't expect any notable performance issues here; we're generally talking about the number of fields on a given model vs keys in the data payload.  _Maybe_ `N*M = ~1000`?

This isn't a crazy common request - it's used to determine RBAC-wise if a user can make changes to a Job Template.

Reply:
I see, thanks for providing more context. About 1000 won't hurt.

Reply:
shell_plus

```python
print [len(cls._meta.concrete_fields) for cls in locals().values() if inspect.isclass(cls) and issubclass(cls, Model) and hasattr(cls, '_meta')]
[11, 51, 6, 4, 13, 17, 18, 4, 10, 6, 11, 5, 18, 28, 19, 11, 3, 3, 8, 5, 9, 8, 6, 26, 11, 4, 2, 54, 4, 45, 41, 18, 6, 12, 3, 2, 11, 13, 15, 11, 16, 7, 10, 5, 33, 5, 8, 2, 40, 12, 5, 6, 22, 35, 3, 9, 29, 11, 38, 14, 16, 36, 3, 3, 3]
```

max 54 -> so about 3,000

Reply:
but it would find its match half-way through on average, so 1,500